### PR TITLE
MINIFICPP-1091 - Updated Windows.md

### PR DESCRIPTION
**Pull request by bakaid**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: Windows.md [hunk 1]
```diff
@@ -17,29 +17,70 @@
...
+| /T | Disables tests |
+| /P | Enables MSI creation |
+| /K | Enables Kafka extension |
+| /J | Enables JNI |
+| /64 | Creates 64-bit build instead of a 32-bit one |
+| /D | Builds RelWithDebInfo build instead of Release |
+
+Examples:
+ - 32-bit build with kafka, disabling tests, enabling MSI creation: `win_build_vs.bat build /T /K /P`
+ - 64-bit build with JNI, with debug symbols: `win_build_vs.bat build /64 /J /D`
```
Comment (arpadboda) [inline Windows.md]: I think our docs shouldn't (even implicitly) recommend building 32 and 64 bit binaries to the same there, so I would prefer changing the 1st arg to "build64" here.

File: Windows.md [hunk 2]
```diff
@@ -17,29 +17,70 @@
...
-The preferred way of building the project is via the win_build_vs.bat script found in our root source folder. You must supply a single command, the build directory. Typically you create a directory
-with CMAKE and build in that referencing your CMAKE tree. Simply supply win_build_vs.bat an argument like 'build' as the name of your build directory and it will create this directory building the 
-project within it. If WiX is installed cpack will create your MSI in the chosen build directory.
+The preferred way of building the project is via the `win_build_vs.bat` script found in our root source folder. Its first parameter is mandatory, the directory in which it will build the project. `build` is a good default choice for this.
+
+After the build directory it will take optional parameters modifying the CMake configuration used in the build:
+
+| Argument | Effect |
+|----------|------------------------------------------------|
+| /T | Disables tests |
```
Comment (arpadboda) [inline Windows.md]: Just nitpicking, this arg behaves the opposite way the other do, which is a bit strange.

File: Windows.md [hunk 3]
```diff
@@ -17,29 +17,70 @@
...
+| /P | Enables MSI creation |
+| /K | Enables Kafka extension |
```
Comment (arpadboda) [inline Windows.md]: Do we have plan to add such cmd line args to enable other extensions as well? 
I like the idea.

File: Windows.md [hunk 4]
```diff
@@ -17,29 +17,70 @@
...
+
+ - Visual Studio 2017
+ - [CMake](https://cmake.org/download/)
+ - [Git](https://git-scm.com/download/win)
+ - (Optional) [WiX Toolset](https://wixtoolset.org/releases/) (only for building the MSI)
+ - (Optional) JDK (only for JNI support)
+
+### JNI support
+Though the project is written in C++, JNI functionality supports running Java processors stored in NiFi Archives. These can be run
+in a much smaller memory footprint and consumer fewer resources. If your systems do not support Java or you do not want a JDK installed, please use non-JNI builds.
```
Comment (arpadboda) [inline Windows.md]: consumer -> consume

File: Windows.md (hunk 2 above)
Comment (bakaid) [reply Windows.md]: Yep, I wouldn't have written this originally, but I don't want to break anyone's workflow who is already using it this way.

File: Windows.md (hunk 3 above)
Comment (bakaid) [reply Windows.md]: CoAP is already in there as /C, just disabled until it actually builds (#687 reenables it), so yes, I think so.

File: Windows.md (hunk 4 above)
Comment (bakaid) [reply Windows.md]: Nice catch, this was originally in the file, I just moved it and didn't verify it properly.

File: Windows.md (hunk 1 above)
Comment (bakaid) [reply Windows.md]: I didn't want to contradict myself by saying that build is a good default choice, but yeah, different names would be safer, I'll change it.

**Pull request closed without merging**